### PR TITLE
Update zwave_js/network_status WS API

### PR DESCRIPTION
**Pull request by raman325**
## Proposed change
Addresses this comment: https://github.com/home-assistant/frontend/pull/12658#discussion_r875548221 in conjunction with the associated core PR

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [x] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- Link to core pull request: https://github.com/home-assistant/core/pull/72205

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Review by zsarnett — DISMISSED** [review]

File: src/data/zwave_js.ts [hunk 1]
```diff
@@ -284,11 +284,13 @@ export const migrateZwave = (
 
 export const fetchZwaveNetworkStatus = (
   hass: HomeAssistant,
-  entry_id: string
+  device_or_entry_id: string,
+  id_type: "entry_id" | "device_id" = "entry_id"
```
Comment (bramkragten) [inline src/data/zwave_js.ts:288]: Can we just make this 2 optional params?
```ts
entry_id?: string,
device_id?: string,
```

```ts
if (!device_id && !entry_id) {
	throw new Error("Either device or entry id should be supplied");
}
```

Comment (bramkragten) [reply src/data/zwave_js.ts:288]: Too make it easier, change the 2 string params to an object that has the 2 string params, so you don't have to pass undefined when you want to use the `device_id` and it helps prevent errors 😄

File: src/panels/config/integrations/integration-panels/zwave_js/dialog-zwave_js-heal-network.ts [hunk 2]
```diff
@@ -204,7 +204,8 @@ class DialogZWaveJSHealNetwork extends LitElement {
     }
     const network: ZWaveJSNetwork = await fetchZwaveNetworkStatus(
       this.hass!,
-      this.entry_id!
+      this.entry_id!,
+      "entry_id"
```
Comment (bramkragten) [inline src/panels/config/integrations/integration-panels/zwave_js/dialog-zwave_js-heal-network.ts:208]:
```suggestion
      {entry_id: this.entry_id!},
```

File: src/panels/config/integrations/integration-panels/zwave_js/dialog-zwave_js-heal-node.ts [hunk 3]
```diff
@@ -204,7 +200,8 @@ class DialogZWaveJSHealNode extends LitElement {
     }
     const network: ZWaveJSNetwork = await fetchZwaveNetworkStatus(
       this.hass!,
-      this.entry_id!
+      this.device!.id,
+      "device_id"
```
Comment (bramkragten) [inline src/panels/config/integrations/integration-panels/zwave_js/dialog-zwave_js-heal-node.ts:204]:
```suggestion
      {device_id: this.device!.id}
```

File: src/panels/config/integrations/integration-panels/zwave_js/zwave_js-config-dashboard.ts [hunk 4]
```diff
@@ -392,7 +392,7 @@ class ZWaveJSConfigDashboard extends LitElement {
 
     const [network, dataCollectionStatus, provisioningEntries] =
       await Promise.all([
-        fetchZwaveNetworkStatus(this.hass!, this.configEntryId),
+        fetchZwaveNetworkStatus(this.hass!, this.configEntryId, "entry_id"),
```
Comment (bramkragten) [inline src/panels/config/integrations/integration-panels/zwave_js/zwave_js-config-dashboard.ts:395]:
```suggestion
        fetchZwaveNetworkStatus(this.hass!, {entry_id: this.configEntryId}),
```

File: src/data/zwave_js.ts [hunk 5]
```diff
@@ -284,14 +284,24 @@ export const migrateZwave = (
...
+  if (!device_or_entry_id.device_id && !device_or_entry_id.entry_id) {
+    throw new Error("Either device or entry ID should be supplied.");
+  }
+  return hass.callWS({
     type: "zwave_js/network_status",
-    device_or_entry_id,
-    id_type,
+    device_or_entry_id:
+      device_or_entry_id.device_id || device_or_entry_id.entry_id,
+    id_type: device_or_entry_id.device_id ? "device_id" : "entry_id",
```
Comment (bramkragten) [inline src/data/zwave_js.ts:302]: Would you not also want to update this on the backend? It is fine by my either way :-)

**Review by bramkragten — DISMISSED** [review]

**Comment by zsarnett** [comment]
Needs to fix the conflicts and then I think we are ready

File: src/data/zwave_js.ts (hunk 5 above)
Comment (raman325) [reply src/data/zwave_js.ts:302]: ah I guess we can do this a little less complicated. I will fix

**Review by zsarnett — APPROVED** [review]

**Pull request merged**